faq/sql-faq.md

```diff
@@ -64,61 +64,7 @@ It is not a bug. The default order of records depends on various situations with
 
 The order of results in MySQL might appear stable because queries are executed in a single thread. However, it is common that query plans can change when upgrading to new versions. It is recommended to use `ORDER BY` whenever an order of results is desired.
 
-The reference can be found in [ISO/IEC 9075:1992, Database Language SQL- July 30, 1992](http://www.contrib.andrew.cmu.edu/~shadow/sql/sql1992.txt), which states as follows:
-
-> If an `<order by clause>` is not specified, then the table specified by the `<cursor specification>` is T and the ordering of rows in T is implementation-dependent.
-
-In the following two queries, both results are considered legal:
-
-```sql
-> select * from t;
-+------+------+
-| a    | b    |
-+------+------+
-|    1 |    1 |
-|    2 |    2 |
-+------+------+
-2 rows in set (0.00 sec)
-```
-
-```sql
-> select * from t; -- the order of results is not guaranteed
-+------+------+
-| a    | b    |
-+------+------+
-|    2 |    2 |
-|    1 |    1 |
-+------+------+
-2 rows in set (0.00 sec)
-```
-
-If the list of columns used in the `ORDER BY` is non-unique, the statement is also considered non-deterministic. In the following example, the column `a` has duplicate values. Thus, only `ORDER BY a, b` is guaranteed deterministic:
-
-```sql
-> select * from t order by a;
-+------+------+
-| a    | b    |
-+------+------+
-|    1 |    1 |
-|    2 |    1 |
-|    2 |    2 |
-+------+------+
-3 rows in set (0.00 sec)
-```
-
-In the following statement, the order of column `a` is guaranteed, but the order of `b` is not guaranteed.
-
-```sql
-> select * from t order by a;
-+------+------+
-| a    | b    |
-+------+------+
-|    1 |    1 |
-|    2 |    2 |
-|    2 |    1 |
-+------+------+
-3 rows in set (0.00 sec)
-```
+You can also use the system variable [`tidb_enable_ordered_result_mode`](/system-variables.md#tidb_enable_ordered_result_mode) to sort the final output result automatically. After setting the variable value to `ON`, TiDB processes `SELECT a, MAX(b) FROM t GROUP BY a` as `SELECT a, MAX(b) FROM t GROUP BY a ORDER BY a, MAX(b)`.
```

Comment:
This is a slightly different point to what it is replacing (and honestly I didn't know what exactly `tidb_enable_ordered_result_mode` stabalized previously because it was not documented). It stabalizes `GROUP BY .. ORDER BY`, so it should be linked from: https://docs.pingcap.com/tidb/stable/mysql-compatibility#limitations-of-select-syntax + the docs for the system variable should also be updated to say that this is only required for 5.7 compatibility, because MySQL 8.0 does not offer order-stability here.

But back to the point of the original FAQ: It is still valid because there are many other statements which are seemingly 'stable' in MySQL. Basically any time you are reading from the clustered index, it will be PK ordered. These are still stable in MySQL 8.0 (and will always be stable when MySQL is single threaded). They are also NOT made stable by `tidb_enable_ordered_result_mode`.

Reply:
`tidb_enable_ordered_result_mode` is implemented by injecting a `Sort` operator to the logical plan. It stabalizes not only `GROUP BY ...` but most of the SELECT statements.

Reply:
@morgo Could you take a look again? Thanks!